### ISSUE TITLE
`Requires.private` in the pkg-config metadata

### ISSUE THREAD

**Issue by nwalfield** [rpm-software-management/rpm-sequoia#36]
There's also the `Requires.private` in the pkg-config metadata, which is currently hardcoded to `nettle`. I'm about to submit a PR to the Fedora rust-rpm-sequoia package with both this patch, and one to just hard-replace `nettle` with `openssl` for their builds... but it'd be cool if there was some easy way to autoconfigure that from `build.rs` as well.

...I'm also wondering, given that RPM is the main consumer for this, whether it wouldn't make sense to generate a CMake configuration (`rpm-sequoia-config.cmake` and `rpm-sequoia-targets.cmake` files) in the build output dir? CMake actually supports linking library targets with no SONAME (you just have to set the `IMPORTED_NO_SONAME` property `TRUE` on the `IMPORTED SHARED` target), so the symlink wouldn't even be necessary to make linking from the builddir work. You'd just do a `cmake -DCMAKE_PREFIX_PATH=/path/to/rpm-sequoia/targets/<profile>/` when configuring CMake, which would use `find_package(rpm-sequoia)` instead of `pkg_check_modules()` to import it.

_Originally posted by @ferdnyc in https://github.com/rpm-software-management/rpm-sequoia/issues/35#issuecomment-1482429051_

**Comment by ferdnyc** [comment]
Heh. Even split off, this is kind of two things. (Bad habit of mine, sorry.)

**Comment by nwalfield** [comment]
> There's also the Requires.private in the pkg-config metadata, which is currently hardcoded to nettle. I'm about to submit a PR to the Fedora rust-rpm-sequoia package with both this patch, and one to just hard-replace nettle with openssl for their builds... but it'd be cool if there was some easy way to autoconfigure that from build.rs as well.

Good point.  I overlooked that.  The backend is behind a feature gate (`sequoia-openpgp/crypto-nettle`, `sequoia-openpgp/crypto-openssl`, etc.)  It would be worth investigating if we could use those to determine how to rewrite that file.  Do you want to do that?
